lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -632,6 +632,11 @@ class BroadcastOpConversionPattern
     mlir::Value input = srcOp.getOperand(0);
     mlir::Value result = srcOp.getResult();
 
+    if (srcOp->getUsers().empty()) {
+      return rewriter.notifyMatchFailure(
+          srcOp, "ttir.broadcast op should have at least one use.");
+    }
+
```

Comment:
Is this actually possible?

Reply:
I found in the mnist example that due to a failure in scalar type conversion, the consumer for broadcast didn't exist, so `rewriter.replaceAllUsesWith(result, input)` after this line failed silently without producing an error. 
The reason for this check is produce an informative message.